### PR TITLE
move loading configmap and secret to common

### PR DESCRIPTION
**Pull request by tinakurian**
moves loading configMap and secret into common. This code will be used in both the host and member operator.

Jira: https://issues.redhat.com/browse/CRT-743
Relates to: https://github.com/codeready-toolchain/toolchain-common/pull/119

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -37,6 +37,7 @@ require (
 
 replace (
 	github.com/Azure/go-autorest => github.com/Azure/go-autorest v13.0.0+incompatible
+	github.com/codeready-toolchain/toolchain-common => github.com/tinakurian/toolchain-common v0.0.0-20200727140128-cfe3823faae8
```
Comment (tinakurian) [inline go.mod:40]: reminder to myself!

**Comment by tinakurian** [comment]
> Looks good, but please consider if storing secrets as env variables is really a safe idea:
> [codeready-toolchain/toolchain-common#119 (comment)](https://github.com/codeready-toolchain/toolchain-common/pull/119#discussion_r461639489)

updated here: 293af5d8d0a092c9dde93f53c7087b19656a5f49

**Comment by MatousJobanek** [comment]
I just realized that you don't print the values from the ConfigMap to the log here: https://github.com/codeready-toolchain/host-operator/blob/master/cmd/manager/main.go#L277-L289
I believe that the values from the ConfigMap are nothing super secret and we can put it into the log - this can be helpful for later troubleshooting. Am I correct @alexeykazakov ?

File: pkg/configuration/configuration.go [hunk 2]
```diff
@@ -97,15 +94,17 @@ const (
 // Config encapsulates the Viper configuration registry which stores the
 // configuration data in-memory.
 type Config struct {
-	host *viper.Viper
+	host         *viper.Viper
+	secretValues map[string][]byte
 }
 
 var log = logf.Log.WithName("configuration")
```
[the remark this answers is not in the document]
Comment (tinakurian) [reply pkg/configuration/configuration.go]: removed here: 469b01986281976dc503b5d84de94303fbfc4531

File: pkg/configuration/configuration.go [hunk 3]
```diff
@@ -3,15 +3,12 @@
...
 	"strings"
 	"time"
 
-	"github.com/operator-framework/operator-sdk/pkg/k8sutil"
+	"github.com/codeready-toolchain/toolchain-common/pkg/configuration"
 	"github.com/spf13/viper"
-	v1 "k8s.io/api/core/v1"
-	"k8s.io/apimachinery/pkg/types"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 	logf "sigs.k8s.io/controller-runtime/pkg/runtime/log"
```
[the remark this answers is not in the document]
Comment (tinakurian) [reply pkg/configuration/configuration.go]: removed here: 469b01986281976dc503b5d84de94303fbfc4531

**Comment by alexeykazakov** [comment]
> I believe that the values from the ConfigMap are nothing super secret and we can put it into the log - this can be helpful for later troubleshooting.

Correct. We should keep secrets in Secrets only. ConfigMaps should not have any secrets. It's fine (and as you said even useful) to print them.

**Comment by MatousJobanek** [comment]
> Correct. We should keep secrets in Secrets only. ConfigMaps should not have any secrets. It's fine (and as you said even useful) to print them.

@tinakurian could you please take care of it?

**Comment by tinakurian** [comment]
> > Correct. We should keep secrets in Secrets only. ConfigMaps should not have any secrets. It's fine (and as you said even useful) to print them.
> 
> @tinakurian could you please take care of it?

done here: 7a418179a8d230d07c2517635bb82d7d0db1f87a

File: cmd/manager/main.go [hunk 4]
```diff
@@ -283,7 +283,9 @@ func printConfig(cfg *configuration.Config) {
 
 	logWithValuesHost := log.WithValues(
 		"key", getHostEnvVarKey(configuration.VarDurationBeforeChangeRequestDeletion),
-		"value", cfg.GetDurationBeforeChangeTierRequestDeletion())
+		"value", cfg.GetDurationBeforeChangeTierRequestDeletion(),
+		"key", getHostEnvVarKey(configuration.VarRegistrationServiceURL),
+		"value", cfg.GetRegistrationServiceURL())
```
Comment (alexeykazakov) [inline cmd/manager/main.go:288]: Two "key"s and two "value"s? It doesn't look right.

Comment (tinakurian) [reply cmd/manager/main.go:288]: @alexeykazakov is this better: https://github.com/codeready-toolchain/host-operator/pull/257/commits/523839e0a56a02047c39b1ae8157524de6c7cc9b#diff-9e63f7b0791edcb4a53c944552b88970L287-L288

**Pull request merged**